packages/yarnpkg-fslib/sources/ZipFS.ts

```diff
@@ -485,16 +489,21 @@ export class ZipFS extends BasePortableFakeFS {
       },
     });
 
+    const fd = this.openSync(p, ``);
+
     const immediate = setImmediate(() => {
       try {
         const data = this.readFileSync(p, encoding);
 
         stream.bytesRead = data.length;
-        stream.write(data);
-        stream.end();
+        stream.end(data);
+        stream.destroy();
```

Comment:
Isn't `destroy` automatically closed on `end`?

Reply:
Interestingly, no. The `close` event is only called when using `destroy`